packages/myst-cli/src/process/file.ts

```diff
@@ -108,9 +110,84 @@ export function loadTexFile(
   return { kind: SourceFileKind.Article, mdast: tex.ast as GenericParent, frontmatter };
 }
 
-export function loadMySTJSON(content: string) {
-  const { mdast, frontmatter, kind } = JSON.parse(content);
-  return { kind: kind ?? SourceFileKind.Article, mdast, frontmatter };
+export function mystJSONValidationOpts(
+  vfile: VFile,
+  opts?: { property?: string; ruleId?: RuleId },
+): ValidationOptions {
+  return {
+    property: opts?.property ?? 'file',
+    file: vfile.path,
+    messages: {},
+    errorLogFn: (message: string) => {
+      fileError(vfile, message, { ruleId: opts?.ruleId ?? RuleId.mystJsonValid });
+    },
+    warningLogFn: (message: string) => {
+      fileWarn(vfile, message, { ruleId: opts?.ruleId ?? RuleId.mystJsonValid });
+    },
+  };
+}
+
+function validateMySTJSON(
+  input: any,
+  opts: ValidationOptions,
+): { mdast: GenericParent; kind: SourceFileKind; frontmatter: PageFrontmatter } {
+  const value = validateObjectKeys(
+    input,
+    { required: ['mdast'], optional: ['kind', 'frontmatter'] },
+    opts,
+  );
+  // Although this is required, we'll set it to an empty article by default
+  const mdast: GenericParent = value?.mdast ?? { type: 'root', children: [] };
+
+  let kind: undefined | SourceFileKind;
+  if (defined(input.kind)) {
+    kind = validateEnum<SourceFileKind>(input.kind, {
+      ...incrementOptions('kind', opts),
+      enum: SourceFileKind,
+    });
+  }
+  let frontmatter: undefined | PageFrontmatter;
+  if (defined(input.frontmatter)) {
+    frontmatter = validatePageFrontmatter(input.frontmatter, incrementOptions('frontmatter', opts));
+  }
+
+  return {
+    mdast,
```

Comment:
Right, I'd like that too. Perhaps we can open that goal down the road - maybe we only build upon the published msyt spec rather than breaking it in main?